qcodes/instrument_drivers/Keysight/keysightb1500/KeysightB1517A.py

```diff
@@ -287,3 +696,256 @@ def use_high_resolution_adc(self) -> None:
+            number: int = 1,
+            mode: constants.AV.Mode = constants.AV.Mode.AUTO
+    ) -> None:
+        """
+        This command sets the number of averaging samples of the high-speed
+        ADC (A/D converter). This command is not effective for the
+        high-resolution ADC. Also, this command is not effective for the
+        measurements using pulse.
+
+        Args:
+            number: 1 to 1023, or -1 to -100. Initial setting is 1.
+                For positive number input, this value specifies the number
+                of samples depended on the mode value.
+                For negative number input, this parameter specifies the
+                number of power line cycles (PLC) for one point measurement.
+                The Keysight B1500 gets 128 samples in 1 PLC.
+                Ignore the mode parameter.
+
+            mode : Averaging mode. Integer expression. This parameter is
+                meaningless for negative number.
+                `constants.AV.Mode.AUTO`: Auto mode (default setting).
+                Number of samples = number x initial number.
+                `constants.AV.Mode.MANUAL`: Manual mode.
+                Number of samples = number
+        """
+        self.write(MessageBuilder().av(number=number, mode=mode).message)
+
+    def connection_mode_of_smu_filter(
+            self,
+            enable_filter: bool,
+            channels: Optional[constants.ChannelList] = None
+    ) -> None:
+        """
+        This methods sets the connection mode of a SMU filter for each channel.
+        A filter is mounted on the SMU. It assures clean source output with
+        no spikes or overshooting. A maximum of ten channels can be set.
+
+        Args:
+            enable_filter : Status of the filter.
+                False: Disconnect (initial setting).
+                True: Connect.
+            channels : SMU channel number. Specify channel from
+                `constants.ChNr` If you do not specify chnum,  the FL
+                command sets the same mode for all channels.
+        """
+        self.write(MessageBuilder().fl(enable_filter=enable_filter,
+                                       channels=channels).message)
+
+    def set_measurement_mode(self,
+                             mode: Union[constants.MM.Mode, int],
+                             channels: Optional[constants.ChannelList] = None
+                             ) -> None:
+        msg = MessageBuilder().mm(mode=mode, channels=channels).message
+        self.write(msg)
+
+    def setup_staircase_sweep(
+            self,
+            v_start: float,
+            v_end: float,
+            n_steps: int,
+            post_sweep_voltage_val: int = constants.WMDCV.Post.STOP,
+            measure_chan_list=(1,),
+            av_coef=-1,
+            enable_filter=True,
+            v_src_range=constants.VOutputRange.AUTO,
+            i_comp=10e-6,
+            i_meas_range=constants.IMeasRange.FIX_10uA,
```

Comment:
If one wants to set different measurement ranges or compliance levels for each channel, how would that be implemented?

Reply:
I am still working on that part. In the current form one cannot do this.

Reply:
@jdwatson I have added a notebook that explains how to set different compliance and measurement ranges.